yt_dlp/extractor/brilliantpala.py

```diff
@@ -0,0 +1,139 @@
+import hashlib
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    dict_get,
+    traverse_obj,
+    urlencode_postdata,
+)
+
+
+class BrilliantpalaIE(InfoExtractor):
+    _NETRC_MACHINE = 'brilliantpala'
+    _DOMAIN = '{subdomain}.brilliantpala.org'
+
+    def _initialize_pre_login(self):
+        self._HOMEPAGE = f'https://{self._DOMAIN}'
+        self._LOGIN_API = f'{self._HOMEPAGE}/login/'
+        self._LOGOUT_DEVICES_API = f'{self._HOMEPAGE}/logout_devices/?next=/'
+        self._CONTENT_API = f'{self._HOMEPAGE}/api/v2.4/contents/{{content_id}}/'
+        self._HLS_AES_URI = f'{self._HOMEPAGE}/api/v2.5/video_contents/{{content_id}}/key/'
+
+    def _get_logged_in_username(self, webpage):
+        if self._LOGIN_API == self._og_search_property('url', webpage):
```

Comment:
I think it would be safer to check the actual url. This goes together with the first suggestion in `BrilliantpalaIE._real_extract()`
```suggestion
    def _get_logged_in_username(self, url, video_id):
        webpage, urlh = self._download_webpage_handle(url, video_id)
        if self._LOGIN_API == urlh.url:
```